### PR TITLE
refactor(@angular-devkit/schematics): remove `UpdateBuffer` and rename `UpdateBuffer2` to `UpdateBuffer`

### PR DESCRIPTION
**Pull request by kyubisation**
This PR removes the internally built `UpdateBuffer` and renames `UpdateBuffer2` (based on magic-string) to `UpdateBuffer`. This should have little to no impact for consumers.

BREAKING CHANGE:
The depracated `UpdateBuffer` has been removed and `UpdateBuffer2` is renamed to `UpdateBuffer`. With this change the related and deprecated symbols `ContentCannotBeRemovedException` and `Chunk` have also been removed.

## PR Checklist

Please check to confirm your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular-cli/blob/main/CONTRIBUTING.md#-commit-message-guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

The deprecated UpdateBuffer instance is being used per default.

Issue Number: N/A

## What is the new behavior?

The magic-string based UpdateBuffer2 is being used per default.

## Does this PR introduce a breaking change?

- [x] Yes
- [ ] No

## Other information

See #21209 and #21110

### CODE REVIEW

**Review by alan-agius4 — CHANGES_REQUESTED** [review]
Thanks for this. 

However, this needs some more work if you are up to it. 

- delete the legacy `UpdateBuffer` 
- rename `UpdateBuffer2` to `UpdateBuffer`

**Comment by kyubisation** [comment]
I was unsure whether to directly delete the UpdateBuffer. If you think it's ok to remove, then I will do that and update it with a breaking change description. I will also remove the `NG_UPDATE_BUFFER_LEGACY` flag, as it would have no effect this way.

**Comment by kyubisation** [comment]
Should I also merge `UpdateBufferBase` with the renamed `UpdateBuffer`?

**Comment by alan-agius4** [comment]
> 



> Should I also merge `UpdateBufferBase` with the renamed `UpdateBuffer`?

No, as `UpdateBufferBase` is part of the public API while `UpdateBuffer` is private.

File: packages/angular_devkit/schematics/src/utility/linked-list.ts [hunk 1]
```diff
@@ -6,6 +6,7 @@
  * found in the LICENSE file at https://angular.io/license
  */
 
+/** @deprecated Since v15.0 */
 export class LinkedList<T extends { next: T | null }> {
```
Comment (alan-agius4) [inline packages/angular_devkit/schematics/src/utility/linked-list.ts:10]: Can be removed since this is private API.

**Review by alan-agius4 — APPROVED** [review]
Thanks for your contribution. One comment about the `LinkedList` otherwise LGTM

File: packages/angular_devkit/schematics/src/utility/linked-list.ts (hunk 1 above)
Comment (kyubisation) [reply packages/angular_devkit/schematics/src/utility/linked-list.ts]: Removed it 👍

**Review by alan-agius4 — APPROVED** [review]

**Pull request merged**